arch/arm/include/rp2040/i2c_slave.h

```diff
@@ -0,0 +1,118 @@
+/****************************************************************************
+ * arch/arm/include/rp2040/i2c_slave.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __ARCH_ARM_INCLUDE_RP2040_I2C_SLAVE_H
+#define __ARCH_ARM_INCLUDE_RP2040_I2C_SLAVE_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/i2c/i2c_slave.h>
+
+#ifndef __ASSEMBLY__
+#if defined(__cplusplus)
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+/* There is no driver of I2C slave operations.  To create an I2C slave,
+ * include this file (as: <arch/chip/i2c_slave.h>)
+ */
+
+/****************************************************************************
+ * Public Type Definitions
+ ****************************************************************************/
+
+typedef int (rp2040_i2c_callback_t)(void *arg, size_t rx_len);
```

Comment:
should we move this typedef  to i2c_slave.h? so we can save the dup typing in many place.

Reply:
That makes sense. I'll do it.